server/routes/testrpc.js

```diff
@@ -0,0 +1,38 @@
+const express = require('express');
+const router = express.Router();
+const exec = require('child_process').exec;
+const config = require('../config.js');
+
+/* Port number for currently running "testrpc" */
+let port = '';
+
+router.post('/start', (req, res) => {
+	const params = req.body.params;
+	
+	let arr = params.split(/ +/);
+	let portIndex = arr.indexOf('-p');
+	if (portIndex === -1) {
+		portIndex = arr.indexOf('--port');
+	}
+	
+	port = (portIndex === -1) ? '8545' : arr[portIndex + 1];
+
+	proc = exec(`${config.testrpcCmd} ${params} | tee ${config.testrpcOut}`, 
```

Comment:
old process not killed here

Reply:
another problem is that if port is not free or the `exec` fails from any other reason, the user will get `success: true`, which is wrong.